libs/langchain/tests/unit_tests/vectorstores/test_utils.py

```diff
@@ -52,3 +56,30 @@ def test_maximal_marginal_relevance_query_dim() -> None:
     first = maximal_marginal_relevance(query_embedding, embedding_list)
     second = maximal_marginal_relevance(query_embedding_2d, embedding_list)
     assert first == second
+
+
+def test_filter_list_metadata() -> None:
+    documents = [
+        Document(
+            page_content="",
+            metadata={
+                "key1": "this is a string!",
```

Comment:
I think worth to add examples of all the default types that should be retained 

Bool
Float
Int
String
None?
Decimal?